Doc/tools/check-warnings.py

```diff
@@ -14,7 +14,7 @@
 from typing import TextIO
 
 # Fail if NEWS nit found before this line number
-NEWS_NIT_THRESHOLD = 200
+NEWS_NIT_THRESHOLD = 1700
```

Comment:
1400 is a suitable limit for this branch: 
https://github.com/python/cpython/actions/runs/9846534114/job/27184520922?pr=121513#step:7:26

```suggestion
NEWS_NIT_THRESHOLD = 1400
```